amd/src/login.js

```diff
@@ -0,0 +1,214 @@
+                $options.dom.password.submit.attr('disabled', 'disabled');
+            }
+
+            // Change state of the submit buttons after input
+            $options.dom.username.input.on('input', $options.dom.username, handleFormInputChange);
+            $options.dom.password.input.on('input', $options.dom.password, handleFormInputChange);
+
+            // Handle change user link
+            $options.dom.password.changeuser.on('click', $options.dom, toggleForms);
+
+            // Handle form submissions
+            $options.dom.username.submit.on('click', $options.dom, $options.username.validation);
+            $options.dom.password.submit.on('click', $options.dom, $options.password.validation);
+        }
+    };
+
+    function handleFormInputChange($event) {
+        var $options = $event.data;
+        $options.input.val() === '' ?
+            $options.submit.attr('disabled', 'disabled') :
+            $options.submit.removeAttr('disabled');
+    }
+
+    function handleUsernameSubmission($event) {
+        $event.preventDefault();
+        var $options = $event.data;
+        var username = $options.username.input.val();
+
+        if (username.toLowerCase() !== 'guest') {
+            setRememberme($options);
+            setReturnurl($options);
+            checkDomain(username, $options);
+        } else {
+            // Populate the username field in the password form
+            $options.password.username.val(username);
+            $options.password.form.submit();
+        }
+    }
+
+    function handlePasswordSubmission($event) {
+        $event.preventDefault();
+        var $options = $event.data;
+        var username = $options.password.username.val();
+
+        queryWebservice(username)[0].done(function(response) {
+            var currentURL = new URL(M.cfg.wwwroot).hostname;
+            if(response.domain !== currentURL) {
+                handleDomainRedirect(response, username);
+            } else {
+                $options.password.form.submit();
+            }
+        });
+    }
+
+    function setRememberme($options) {
+        var $remFromUserForm = $options.username.rememberme;
+        var $remFromPassForm = $options.password.rememberme;
+        if ($remFromUserForm.prop('checked')) {
+            $remFromPassForm.val("1");
+        }
+    }
+
+    function setReturnurl($options) {
+        var retFromUserFormVal = $options.username.returnurl.val();
+        var $retFromPassForm = $options.password.returnurl;
+        if (retFromUserFormVal) {
+            $retFromPassForm.val(retFromUserFormVal);
+        }
+    }
+
+    function checkDomain(username, $options) {
+        var result = false;
+        queryWebservice(username)[0].done(function(response) {
+            result = handleDomainRedirect(response, username, $options);
+        }).fail(function() {
+            get_string_and_notify('danger', 'invalid_user');
+            result = false;
+        });
+        return result;
```

Comment:
This return value is unused and can be removed.

Reply:
The return was in there so that in the event that an invalid username was passed the scope returned to the 'sumbitusername' function. The only thing this currently effects is the above taking place and then the user trying access the site using 'guest' which will not be checked if the return is removed. Even though the chance of this happening is low, i feel it is better to still account for this.

Reply:
If you're not using the return value and just want to break execution, return nothing:

```javascript
return;
```